test/conformance/parallel/ptp.go

```diff
@@ -0,0 +1,271 @@
+				} else if podRole(&ptpPods.Items[podIndex], utils.PtpGrandmasterNodeLabel) {
+					waitUntilLogIsDetected(&ptpPods.Items[podIndex], utils.TimeoutIn3Minutes, "Profile Name:")
+					ptpMasterRunningPods = append(ptpMasterRunningPods, ptpPods.Items[podIndex])
+				}
+			}
+			if testconfig.GlobalConfig.DiscoveredGrandMasterPtpConfig != nil {
+				Expect(len(ptpMasterRunningPods)).To(BeNumerically(">=", 1), "Fail to detect PTP master pods on Cluster")
+				Expect(len(ptpSlaveRunningPods)).To(BeNumerically(">=", 1), "Fail to detect PTP slave pods on Cluster")
+			} else {
+				Expect(len(ptpSlaveRunningPods)).To(BeNumerically(">=", 1), "Fail to detect PTP slave pods on Cluster")
+			}
+			//ptpRunningPods = append(ptpMasterRunningPods, ptpSlaveRunningPods...)
+		})
+
+		It("test-feature-1", func() {
+			done := make(chan interface{})
+			go func() {
+				logrus.Info("Testing feature 1")
+				time.Sleep(100 * time.Millisecond)
+				logrus.Info("Ending testing feature 1")
+				close(done)
+			}()
+			Eventually(done, utils.TimeoutIn5Minutes).Should(BeClosed())
+		})
+		It("test-feature-2", func() {
+			done := make(chan interface{})
+			go func() {
+				logrus.Info("Testing feature 2")
+				time.Sleep(500 * time.Millisecond)
+				logrus.Info("Ending testing feature 2")
+				close(done)
+			}()
+			Eventually(done, utils.TimeoutIn10Minutes).Should(BeClosed())
+		})
+
+		It("test-feature-3", func() {
+			done := make(chan interface{})
+			go func() {
+				logrus.Info("Testing feature 3", 0)
+				time.Sleep(500 * time.Millisecond)
+				logrus.Info("Ending testing feature 3")
+				close(done)
+			}()
+			Eventually(done, utils.TimeoutIn3Minutes).Should(BeClosed())
+			Expect("banashri").ToNot(BeNil())
+		})
+
+		It("continuous-offset-testing", func() {
+
+			Expect(testclient.Client).NotTo(BeNil())
+
+			logrus.Debug("soak-testing started")
+
+			logrus.Info("config=", testParameters)
+			//
+			if fullConfig.Status == testconfig.DiscoveryFailureStatus {
+				Fail("Failed to find a valid ptp slave configuration")
+			}
+			logrus.Infof("testclient: %v", testclient.Client)
+			// Get All PTP pods
+			slaveNodes, err := testclient.Client.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{
+				LabelSelector: utils.PtpClockUnderTestNodeLabel,
+			})
+			if err != nil {
+				logrus.Error("Can't list slave nodes")
+				Fail("Can't list slave nodes")
+			}
+
+			var slavePods []v1.Pod
+
+			for _, s := range slaveNodes.Items {
+				ptpPods, err := testclient.Client.CoreV1().Pods(utils.PtpLinuxDaemonNamespace).List(context.Background(),
+					metav1.ListOptions{LabelSelector: "app=linuxptp-daemon", FieldSelector: fmt.Sprintf("spec.nodeName=%s", s.Name)})
+				if err != nil {
+					logrus.Error("Error in getting ptp pods")
+					Fail("can't find ptp pods, test skipped")
+				}
+				slavePods = append(slavePods, ptpPods.Items...)
+			}
+
+			if len(slavePods) == 0 {
+				logrus.Error("No slave pod found")
```

Comment:
all error and log messages should be lowercase

Reply:
Changed